control-plane/csi-driver/src/bin/controller/client.rs

```diff
@@ -123,16 +123,50 @@ impl RestApiClient {
         let url = clients::tower::Url::parse(endpoint)
             .map_err(|error| anyhow!("Invalid API endpoint URL {}: {:?}", endpoint, error))?;
         let concurrency_limit = cfg.create_volume_limit() * 2;
-        let tower = clients::tower::Configuration::builder()
-            .with_timeout(cfg.io_timeout())
-            .with_concurrency_limit(Some(concurrency_limit))
-            .build_url(url)
-            .map_err(|error| {
-                anyhow::anyhow!(
-                    "Failed to create openapi configuration, Error: '{:?}'",
-                    error
-                )
-            })?;
+        let ca_certificate_path = cfg.ca_certificate_path();
+        let cert = match ca_certificate_path {
+            Some(path) => {
+                let cert = std::fs::read(path).expect("Failed to read certificate file");
+                Some(cert)
+            },
+            None => None,
+        };
+        let tower = match (url.scheme(), cert) {
+            ("https", Some(cert)) => {
+                debug!("Attempting TLS connection to {}", url);
+
+                // Use new_with_client method to create the configuration
+                clients::tower::Configuration::builder()
+                    .with_timeout(Some(cfg.io_timeout()))
+                    .with_concurrency_limit(Some(concurrency_limit))
+                    .with_certificate(cert.as_slice())
+                    .build_url(url)
+                    .map_err(|error| {
+                        anyhow::anyhow!(
+                            "Failed to create openapi configuration, Error: '{:?}'",
+                            error
+                        )
+                    })?
+            },
+            ("https", None) => {
+                anyhow::bail!("HTTPS endpoint requires a CA certificate path");
```

Comment:
Should we keep this as an option? This is what we were thinking of enabling as a first step.
Granted the existing certificate is a hardcoded dummy, but would already be better than plain HTTP as a default option?

Reply:
If certs aren't provided, we shouldn't be using https as server can't be configured without certs.  It makes more sense to me to not rely on dummy certs at all and I would lobby for cert-manager to be installed and certs created if tls is enabled and remove any reliance on dummy certs.  That's something that needs to be decided- should cert-manager be used? -what are the benefits and drawbacks of dependency on cert-manager and if not used how will users generate certs?

Reply:
https://github.com/openebs/openebs/pull/3856